src/box/txn.c

```diff
@@ -1030,7 +1031,7 @@ txn_prepare(struct txn *txn)
 	if (txn_check_can_continue(txn) != 0)
 		return -1;
 
-	if (txn->rollback_timer != NULL) {
+	if (!box_begin_timeout_new_meaning && txn->rollback_timer != NULL) {
```

Comment:
This looks confusing. Why with the old compat it works here and with the new one, several lines below. What is the reason for that? Is it a bug? If so, we should just move it below and fix in the separate commit

Reply:
In the old behavior, the timeout did not cover `before_commit` triggers.
In the new behavior, the timeout covers the `before_commit` trigger and the entire commit after it. And when discussing the design, we decided that the transaction would be rolled back if the timeout expired while the triggers were executing.